### ISSUE TITLE
About the prediction accuracy after training

### ISSUE THREAD

**Issue by XinGP** [HKUST-Aerial-Robotics/SIMPL#9]
![27131f5966bf59c70a2673ecb21bf3e](https://github.com/HKUST-Aerial-Robotics/SIMPL/assets/53513246/09cdd0ca-8119-4bdb-82b7-8418f66aced8)
Hello author! I reproduced your AV1 training code, adjusted the epoch to 50, and adjusted the batch_size to 4, with no other adjustments made. What is the reason for the significant difference in the validation results obtained after training compared to your validation model?

**Comment by MasterIzumi** [comment]
@XinGP Generally, if you reduce the batch size, you should also reduce the learning rate (ref: https://arxiv.org/pdf/1812.01187.pdf). The relatively large LR will lead to unstable training.

**Comment by penglo** [comment]
> ![27131f5966bf59c70a2673ecb21bf3e](https://private-user-images.githubusercontent.com/53513246/323873796-09cdd0ca-8119-4bdb-82b7-8418f66aced8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY1NDc5MTAsIm5iZiI6MTcxNjU0NzYxMCwicGF0aCI6Ii81MzUxMzI0Ni8zMjM4NzM3OTYtMDljZGQwY2EtODExOS00YmRiLTgyYjctODQxOGY2NmFjZWQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI0VDEwNDY1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNDk5NmVjNmQ5NTZiYTA0ZDY1OWNmMmViZmJjYmY4NTM0YWNmZmEzYzQ2OGQwMzkzYTlkZTg2NzM0ZjA0NjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KatS9ymQFLqCW73LYg1nopmXo0YO-jPalgSHPn7jsmw) Hello author! I reproduced your AV1 training code, adjusted the epoch to 50, and adjusted the batch_size to 4, with no other adjustments made. What is the reason for the significant difference in the validation results obtained after training compared to your validation model?
Hello, I've also encountered similar issues while tuning parameters, and I can't pinpoint the specific learning rate. Could you share what learning rate you ended up using? I'd like to discuss this issue further. My email is lipl23@mails.jlu.edu.cn.

**Comment by Family-Liao** [comment]
Have you found a suitable learning rate for batch_size 4 to close to the result which the author gives?

**Comment by Family-Liao** [comment]
> ![27131f5966bf59c70a2673ecb21bf3e](https://private-user-images.githubusercontent.com/53513246/323873796-09cdd0ca-8119-4bdb-82b7-8418f66aced8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczNzYzODcsIm5iZiI6MTcxNzM3NjA4NywicGF0aCI6Ii81MzUxMzI0Ni8zMjM4NzM3OTYtMDljZGQwY2EtODExOS00YmRiLTgyYjctODQxOGY2NmFjZWQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAzVDAwNTQ0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkNjhkY2U3YTI4MjkwMmQ1OGY4MWZlNTA2OTk4ZDQ3MmUyZmM2OWNlNzIxNzM1YzZmNmIzOGQ5MjE5NWRiMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OCMaWgX8Hls6j_DnlDxpOlzBHHMf9i1iMQju6R2YsJA) Hello author! I reproduced your AV1 training code, adjusted the epoch to 50, and adjusted the batch_size to 4, with no other adjustments made. What is the reason for the significant difference in the validation results obtained after training compared to your validation model?
Have you found a suitable learning rate for batch_size 4 to close to the result which the author gives?

**Comment by XinGP** [comment]
> @XinGP一般来说，如果减小batch size，也应该减小学习率（ref: https://arxiv.org/pdf/1812.01187.pdf）。LR过大会导致训练不稳定。

Yes, after reducing all learning rates by half, I can obtain validation set results similar to those in the paper

**Comment by Family-Liao** [comment]
> > [@XinGP](https://github.com/XinGP)一般来说，如果减小batch size，也应该减小学习率（ref: [https://arxiv.org/pdf/1812.01187.pdf）。LR过大会导致训练不稳定。](https://arxiv.org/pdf/1812.01187.pdf%EF%BC%89%E3%80%82LR%E8%BF%87%E5%A4%A7%E4%BC%9A%E5%AF%BC%E8%87%B4%E8%AE%AD%E7%BB%83%E4%B8%8D%E7%A8%B3%E5%AE%9A%E3%80%82)
> 
> Yes, after reducing all learning rates by half, I can obtain validation set results similar to those in the paper

请问，将学习率全部减半之后，你的批次大小调整为多大了 @XinGP

**Comment by XinGP** [comment]
可以参考我的代码库https://github.com/XinGP/DGFNet中的config配置文件，这个工作是基于simpl进行改进的。opt_cfg['scheduler'] == 'polyline':
            opt_cfg['init_lr'] = 1e-5
            opt_cfg['milestones'] = [0, 5, 25, 40]
            opt_cfg['values'] = [5e-5, 5e-4, 3e-4, 5e-5]，batch_size是16，一台3090进行训练就可以达到和simpl差不多的结果

**Comment by XinGP** [comment]
可以参考我的代码库   https://github.com/XinGP/DGFNet  中的config配置文件，这个工作是基于simpl进行改进的。batch_size是16，一台3090进行训练就可以达到和simpl差不多的结果。

**Comment by XinGP** [comment]
elif opt_cfg['scheduler'] == 'polyline':
opt_cfg['init_lr'] = 1e-5
opt_cfg['milestones'] = [0, 5, 35, 40]
opt_cfg['values'] = [5e-5, 5e-4, 3e-4, 1e-4]

**Comment by Family-Liao** [comment]
非常感谢您的详细回复，请问这个学习率对应的批大小是16吗？还有，看到您回复到了，您基于Simpl做了改进，请问是否能把测试集相关的代码（生成test预处理文件和生成h5文件上传到榜单上的）发我一份供我学习吗？希望能得到您的回复，再次感谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HKUST-Aerial-Robotics/SIMPL"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月17日(星期一) 下午5:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HKUST-Aerial-Robotics/SIMPL] About the prediction accuracy after training (Issue #9)



   
elif opt_cfg['scheduler'] == 'polyline':
 opt_cfg['init_lr'] = 1e-5
 opt_cfg['milestones'] = [0, 5, 35, 40]
 opt_cfg['values'] = [5e-5, 5e-4, 3e-4, 1e-4]

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***&gt;
  XinGP left a comment (HKUST-Aerial-Robotics/SIMPL#9)
 
elif opt_cfg['scheduler'] == 'polyline':
 opt_cfg['init_lr'] = 1e-5
 opt_cfg['milestones'] = [0, 5, 35, 40]
 opt_cfg['values'] = [5e-5, 5e-4, 3e-4, 1e-4]

**Comment by XinGP** [comment]
是的我当时也是改了simpl的batch到16，然后调整成我发的学习率，测试集代码可以参考https://github.com/XinGP/DGFNet里的test.py文件。h5文件可以留下你的邮箱，我发给你。

**Comment by XinGP** [comment]
就是我当时也修改了简单的batch到16，然后调整成我发的学习率，测试集代码可以参考 https://github.com/XinGP/DGFNet 里的test.py文件。h5文件可以留下你的邮箱，我发给你。
